internal/node/node.go

```diff
@@ -72,3 +76,45 @@ func (n *node) FindNodeCondition(cond []v1.NodeCondition, conditionType v1.NodeC
 
 	return nil
 }
+
+func (n *node) UpdateLabels(ctx context.Context, node *v1.Node, toBeAdded, toBeRemoved []string) error {
+	patchFrom := client.MergeFrom(node.DeepCopy())
+
+	n.nh.addLabels(node, toBeAdded)
+	n.nh.removeLabels(node, toBeRemoved)
+
+	if err := n.client.Patch(ctx, node, patchFrom); err != nil {
+		return fmt.Errorf("could not patch node: %v", err)
+	}
+	return nil
+}
+
+type nodeHelperImpl struct{}
+
+type nodeHelper interface {
+	addLabels(node *v1.Node, labels []string)
+	removeLabels(node *v1.Node, labels []string)
+}
```

Comment:
I am not sure a helper interface is the best option here. I would make them non-exported functions (not methods) in the package.

Here are some points to consider when deciding if something should be an interface/function.
1. As a general rule of thumb, if a method isn't using its "self" object then it should be a function.
2. If an API is exported, using its "self" object and wish to be mocked in unit-tests then we use an exported interface.
3. In some cases, we are having exceptions in the form of "helper, non-exported interfaces" - in cases where a packages has a lot of non-exported functions/methods we sometimes choose to put them inside an helper interface (even if the "self" object isn't being used just for the ease of unit-testing the package because it becomes difficult to test.

In this case, `addLabels` and `removeLabels`:
1. Does not use the "self" (node) object - leaning towards being functions rather than methods.
2. Are pretty short/simple and there are only 2 "helper" functions so it is not too difficult to unit-test `UpdateLabels`

Therefore I would make those non-exported functions and remove the helper interface in this package.

Reply:
@ybettan  so in that case, you think that `addLabels` and `removeLabels` should not be mocked? 
they are pretty simple, right, but mocking them helps for example in line 173 in node_test.go

Reply:
deleted the `nodeHelper` interface

Reply:
> so in that case, you think that addLabels and removeLabels should not be mocked?

Indeed. In this case I don't think the mocking benefit worth writing another interface.